ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.component.js

```diff
@@ -0,0 +1,84 @@
+import React, { useContext } from 'react';
+
+import PropTypes from 'prop-types';
+import { I18nContext } from '../../../contexts/i18n';
+
+import ToggleButton from '../../ui/toggle-button';
+import { Box, Text } from '../../component-library';
+import {
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+  FontWeight,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+
+import NetworkToggle from './NetworkToggle';
+import { useIncomingTransactionToggle } from './useIncomingTransactionToggle';
+
+const IncomingTransactionToggle = ({
+  wrapperRef,
+  incomingTransactionsPreferences,
+  allNetworks,
+  setIncomingTransactionsPreferences,
+}) => {
+  const t = useContext(I18nContext);
+
+  const {
+    networkPreferences,
+    isAllEnabled,
+    toggleAllEnabled,
+    toggleSingleNetwork,
+  } = useIncomingTransactionToggle({
+    incomingTransactionsPreferences,
+    allNetworks,
+    setIncomingTransactionsPreferences,
+  });
+
+  return (
+    <Box ref={wrapperRef}>
```

Comment:
Can we add a `className` here? Since this component doesn't have a story, adding a className will be helpful for other engineers to look for this component in future.

Reply:
yes definitely ! good idea ~